### PR TITLE
feat(RHINENG-16842) Include hosts in "ungrouped" group when filtering on group=None

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-16842](https://issues.redhat.com/browse/RHINENG-16842).
In order for HBI to work in read-only mode during the Kessel Phase 0 migration, we need to allow the None attributeFilter value from RBAC to not only allow access to ungrouped hosts, but also hosts that are in the "ungrouped" group. This will allow the API to continue working as expected in-between HBI's data migration and RBAC's data migration.

Also, mgmt-fabric doesn't necessarily need to remove the None attributeFilter values during migration, and it may exist afterwards. Because of this, I'm removing the code that restricted the None value post-migration.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by jpramos123 — APPROVED** [review]
LGTM!

File: lib/host_repository.py [hunk 1]
```diff
@@ -376,13 +376,15 @@ def get_host_list_by_id_list_from_db(host_id_list, identity, rbac_filter=None, c
     )
     if rbac_filter and "groups" in rbac_filter:
         rbac_group_filters = (HostGroupAssoc.group_id.in_(rbac_filter["groups"]),)
-        # Special handling for "None" rbac attributeFilter, no longer needed after Kessel migration
-        if None in rbac_filter["groups"] and not get_flag_value(FLAG_INVENTORY_KESSEL_WORKSPACE_MIGRATION):
-            rbac_group_filters += (HostGroupAssoc.group_id.is_(None),)
+        if None in rbac_filter["groups"]:
```
Comment (jpramos123) [inline lib/host_repository.py:379]: Why removing the feature flag?

Comment (kruai) [reply lib/host_repository.py:379]: We had originally planned to disallow the "None" attributeFilter value post-migration, but they've decided to leave it for now, which means we need to continue to support it.
